### PR TITLE
[DataObject] Fix deleting folders

### LINKED ISSUE #6945: Can not delete folders in version 6.7.0
# Bug Report

I updated PimCore to the newest version 6.7.0 today and tried to delete a folder. This issue is reproducable on the demo site.

### Actual behavior

Deleting a folder gives you the following error:
`Call to undefined method getClassName in class Pimcore\Model\DataObject\Folder`

### Steps to reproduce

Create a folder or select an existing folder and try to delete it by opening the right mouse button menu.

### Trace

```json
 [{
 	"file": "\/var\/www\/pimcore\/vendor\/pimcore\/pimcore\/models\/DataObject\/AbstractObject.php",
 	"line": 1350,
 	"function": "__call",
 	"class": "Pimcore\\Model\\AbstractModel",
 	"type": "-\u003E",
 	"args": ["getClassName", []]
 }, {
 	"file": "\/var\/www\/pimcore\/vendor\/pimcore\/pimcore\/models\/DataObject\/AbstractObject.php",
 	"line": 361,
 	"function": "__call",
 	"class": "Pimcore\\Model\\DataObject\\AbstractObject",
 	"type": "-\u003E",
 	"args": ["getClassName", []]
 }, {
 	"file": "\/var\/www\/pimcore\/vendor\/pimcore\/pimcore\/bundles\/AdminBundle\/Controller\/Admin\/ElementControllerBase.php",
 	"line": 160,
 	"function": "getList",
 	"class": "Pimcore\\Model\\DataObject\\AbstractObject",
 	"type": "::",
 	"args": [{
 		"unpublished": true
 	}]
 }, {
 	"file": "\/var\/www\/pimcore\/vendor\/pimcore\/pimcore\/bundles\/AdminBundle\/Controller\/Admin\/DataObject\/DataObjectController.php",
 	"line": 92,
 	"function": "deleteInfoAction",
 	"class": "Pimcore\\Bundle\\AdminBundle\\Controller\\Admin\\ElementControllerBase",
 	"type": "-\u003E",
 	"args": [{
 		"attributes": {},
 		"request": {},
 		"query": {},
 		"server": {},
 		"files": {},
 		"cookies": {},
 		"headers": {}
 	}]
 }, {
 	"file": "\/var\/www\/pimcore\/vendor\/symfony\/symfony\/src\/Symfony\/Component\/HttpKernel\/HttpKernel.php",
 	"line": 158,
 	"function": "deleteInfoAction",
 	"class": "Pimcore\\Bundle\\AdminBundle\\Controller\\Admin\\DataObject\\DataObjectController",
 	"type": "-\u003E",
 	"args": [{
 		"attributes": {},
 		"request": {},
 		"query": {},
 		"server": {},
 		"files": {},
 		"cookies": {},
 		"headers": {}
 	}]
 }, {
 	"file": "\/var\/www\/pimcore\/vendor\/symfony\/symfony\/src\/Symfony\/Component\/HttpKernel\/HttpKernel.php",
 	"line": 80,
 	"function": "handleRaw",
 	"class": "Symfony\\Component\\HttpKernel\\HttpKernel",
 	"type": "-\u003E",
 	"args": [{
 		"attributes": {},
 		"request": {},
 		"query": {},
 		"server": {},
 		"files": {},
 		"cookies": {},
 		"headers": {}
 	}, 1]
 }, {
 	"file": "\/var\/www\/pimcore\/vendor\/symfony\/symfony\/src\/Symfony\/Component\/HttpKernel\/Kernel.php",
 	"line": 201,
 	"function": "handle",
 	"class": "Symfony\\Component\\HttpKernel\\HttpKernel",
 	"type": "-\u003E",
 	"args": [{
 		"attributes": {},
 		"request": {},
 		"query": {},
 		"server": {},
 		"files": {},
 		"cookies": {},
 		"headers": {}
 	}, 1, true]
 }, {
 	"file": "\/var\/www\/pimcore\/web\/app.php",
 	"line": 36,
 	"function": "handle",
 	"class": "Symfony\\Component\\HttpKernel\\Kernel",
 	"type": "-\u003E",
 	"args": [{
 		"attributes": {},
 		"request": {},
 		"query": {},
 		"server": {},
 		"files": {},
 		"cookies": {},
 		"headers": {}
 	}]
 }]
```

### PR DESCRIPTION
**Pull request by dvesh3**
## Changes in this pull request  
Resolves #6945

### CODE REVIEW

File: models/DataObject/AbstractObject.php [hunk 1]
```diff
@@ -355,7 +355,7 @@ public static function getList($config = [])
     {
         $className = DataObject::class;
         // get classname
-        if (!in_array(static::class, [__CLASS__, Concrete::class], true)) {
+        if (!in_array(static::class, [__CLASS__, Concrete::class, Folder::class], true)) {
```
Comment (ssiergl) [inline models/DataObject/AbstractObject.php:358]: @dvesh3 well this is a problem if a bundle using model class overrides (eg overriding Folder class extending original Pimcore Folder class will make this condition truthy).

**Review by ssiergl — COMMENTED** [review]
see inline comment

File: models/DataObject/AbstractObject.php (hunk 1 above)
Comment (NiklasBr) [reply models/DataObject/AbstractObject.php:358]: Would `self::OBJECT_TYPE_FOLDER` work?

Comment (ssiergl) [reply models/DataObject/AbstractObject.php:358]: No...getType isn't static

should be something like this

```
        // get classname
        if (!in_array(static::class, [__CLASS__, Concrete::class, Folder::class], true)) {
            /** @var Concrete $tmpObject */
            $tmpObject = new static();
            if ($tmpObject instanceof Concrete) {
                $className = 'Pimcore\\Model\\DataObject\\' . ucfirst($tmpObject->getClassName());
            }
        }
```

Comment (dvesh3) [reply models/DataObject/AbstractObject.php:358]: thanks @ssiergl I'll go with your suggested fix. Would you like to provide a PR?

Comment (ssiergl) [reply models/DataObject/AbstractObject.php:358]: @dvesh3 will do this evening

**Pull request merged**